src/yetibot/core/adapters/irc.clj

```diff
@@ -64,10 +70,7 @@
         (catch java.net.SocketException e
           ; it must have disconnect, try reconnecting again
           ; TODO add better retry, like Slack
-          (log/info "SocketException, trying to reconnect in" wait-before-reconnect "ms")
-          (Thread/sleep wait-before-reconnect)
-          (connect adapter)
```

Comment:
I think `connect` here is a typo and should be `stop` instead, because `start` does `connect` inside.
Otherwise this happen:
- `(connect adapter)` starts connecting
- irclj sends `NICK` command
- few milliseconds later `(start adapter)` calls `(connect)` again and replaces `conn` atom
- irclj creates new connection and sends `NICK` 
- nick already in use received